controllers/bounce_processes.go

```diff
@@ -251,27 +253,29 @@ func getProcessesReadyForRestart(logger logr.Logger, cluster *fdbv1beta2.Foundat
 		}
 	}
 
-	// If we upgrade the cluster wait until all processes are ready for the restart. We don't want to block the restart
-	// if some processes are already upgraded e.g. in the case of version compatible upgrades and we also don't want to
-	// block the restart command if a process is missing longer than the specified GetIgnoreMissingProcessesSeconds.
-	// Those checks should ensure we only run the restart command if all processes that have to be restarted and are connected
-	// to cluster are ready to be restarted.
-	expectedProcesses := counts.Total() - missingProcesses + markedForRemoval
-	// If more than one storage server per Pod is running we have to account for this. In this case we have to add the
-	// additional storage processes.
-	if cluster.Spec.StorageServersPerPod > 1 {
-		expectedProcesses += counts.Storage * (cluster.Spec.StorageServersPerPod - 1)
-	}
+	if cluster.IsBeingUpgradedWithVersionIncompatibleVersion() {
+		// If we upgrade the cluster wait until all processes are ready for the restart. We don't want to block the restart
+		// if some processes are already upgraded e.g. in the case of version compatible upgrades and we also don't want to
+		// block the restart command if a process is missing longer than the specified GetIgnoreMissingProcessesSeconds.
+		// Those checks should ensure we only run the restart command if all processes that have to be restarted and are connected
+		// to cluster are ready to be restarted.
+		expectedProcesses := counts.Total() - missingProcesses + markedForRemoval
+		// If more than one storage server per Pod is running we have to account for this. In this case we have to add the
+		// additional storage processes.
```

Comment:
It would be nice if we could push this logic into "GetProcessCountsWithDefaults()".

Reply:
I create an issue to look into this. We cannot move the code as it is directly into the `GetProcessCountsWithDefaults` method as this method is used in different places to get the number of process groups, so probably we need to implement a new method.

Reply:
https://github.com/FoundationDB/fdb-kubernetes-operator/issues/1804